src/diff/children.js

```diff
@@ -228,6 +233,35 @@ export function diffChildren(
 	}
 }
 
+function reorderChildren(childVNode, oldDom, parentDom) {
+	for (let tmp = 0; tmp < childVNode._children.length; tmp++) {
+		const vnode = childVNode._children[tmp];
+		if (vnode) {
+			vnode._parent = childVNode;
+
+			if (typeof vnode.type == 'function') {
+				reorderChildren(vnode, oldDom, parentDom);
```

Comment:
We'd need to find a way to tell whether or not we have to traverse further:

```
const Cell = () => (
  <Fragment>
    <div>x</div>
    <Fragment><div>y</div></Fragment>
  </Fragment>
)
```

When this Cell bails we need to ensure we move `y` as well as `x` when this is moved from the top. Before this change we would forget about `y` within `diffChildren` but move it from `diff.shouldComponentUpdate = false`.